### PR TITLE
PHP: Changed the way to parse an array type by analogy with other types.

### PR DESCRIPTION
**Pull request by troizet**
What was done in this PR:
- Changed the way to parse an array type by analogy with other types
- Fixed broken tests
- Fixed issue #5375
- New tests for fixed issue have been added
- Fixed broken `UnusableTypesHintError`

The reason for issue #5375 is that the condition: https://github.com/apache/netbeans/blob/80ee045a8fb276323d6f756c8947950a07dbf540/php/php.editor/src/org/netbeans/modules/php/editor/parser/api/Utils.java#L106-L108 is not fulfilled, because the array type is parsed differently than the other type.
In this PR I tried to change the parser so that the array type is parsed similarly to other types.
Locally, all unit tests passed successfully.
Manual testing showed no problems either.

### CODE REVIEW

**Comment by troizet** [comment]
Somehow missed the broken test. Sorry. I'll keep looking into it.

File: php/php.editor/src/org/netbeans/modules/php/editor/verification/UnusableTypesHintError.java [hunk 1]
```diff
@@ -590,7 +587,7 @@ private static boolean isCallableType(Identifier identifier) {
         private static boolean isVoidType(NamespaceName namespaceName) {
             return Type.VOID.equals(CodeUtils.extractUnqualifiedName(namespaceName));
         }
-
+
```
Comment (junichi11) [inline php/php.editor/src/org/netbeans/modules/php/editor/verification/UnusableTypesHintError.java:590]: Please remove whitespaces.

File: php/php.editor/src/org/netbeans/modules/php/editor/verification/UnusableTypesHintError.java [hunk 2]
```diff
@@ -576,11 +574,10 @@ private void createRedundantTypeCombinationError(ASTNode node, UnionType unionTy
         private void createIterableRedundantTypeCombinationError(UnionType unionType, IterableRedundantTypeCombination.RedundantType redundantType) {
             hints.add(new IterableRedundantTypeCombination(rule, fileObject, unionType.getStartOffset(), unionType.getEndOffset(), unionType, redundantType));
         }
-
-        private static boolean isArrayType(Identifier identifier) {
-            return !identifier.isKeyword()
-                    && Type.ARRAY.equals(identifier.getName().toLowerCase(Locale.ENGLISH));
-        }
+
```
Comment (junichi11) [inline php/php.editor/src/org/netbeans/modules/php/editor/verification/UnusableTypesHintError.java:577]: Please remove a whitespace.

File: php/php.editor/src/org/netbeans/modules/php/editor/verification/UnusableTypesHintError.java (hunk 1 above)
Comment (troizet) [reply php/php.editor/src/org/netbeans/modules/php/editor/verification/UnusableTypesHintError.java]: Done.

File: php/php.editor/src/org/netbeans/modules/php/editor/verification/UnusableTypesHintError.java (hunk 2 above)
Comment (troizet) [reply php/php.editor/src/org/netbeans/modules/php/editor/verification/UnusableTypesHintError.java]: Done.

**Review by junichi11 — APPROVED** [review]
Thanks!

File: php/php.editor/tools/ASTPHP5Parser.cup [hunk 3]
```diff
@@ -2625,10 +2625,9 @@ fully_qualified_class_name:className
     RESULT = className;
 :}
 
-| T_ARRAY:array
+| T_ARRAY:name
 {:
-    Identifier classId = new Identifier(arrayleft, arrayright, "array");
-    RESULT = classId;
+    RESULT = NamespaceName.create(nameleft, nameright, name);
```
Comment (tmysik) [inline php/php.editor/tools/ASTPHP5Parser.cup:2630]: @junichi11 Is this really correct? I am sorry, but I simply cannot recall it... (yes, it is a long time ago, already :)

Thanks.

**Review by tmysik — APPROVED** [review]

**Comment by tmysik** [comment]
@junichi11 Please, see my question above. If it is OK, feel free to merge this PR. Thank you.

**Comment by tmysik** [comment]
@troizet Thanks a lot for this PR! It is really great to have it!

File: php/php.editor/tools/ASTPHP5Parser.cup (hunk 3 above)
Comment (junichi11) [reply php/php.editor/tools/ASTPHP5Parser.cup:2630]: `string` is recognized as a namespace name with #4037
https://github.com/apache/netbeans/blob/1d85a3e854fafef563a3b0986b88f4e9613a1d85/php/php.editor/tools/ASTPHP5Parser.cup#L4323-L4348

So, this should be OK, I think.
If we come across a problem, let's fix it. Thanks!

Comment (tmysik) [reply php/php.editor/tools/ASTPHP5Parser.cup:2630]: Thanks!

**Pull request merged**